src/data/fun_like/basic.lean

```diff
@@ -128,6 +129,11 @@ class fun_like (F : Sort*) (α : out_param Sort*) (β : out_param $ α → Sort*
 (coe : F → Π a : α, β a)
 (coe_injective' : function.injective coe)
 
+instance subtype.fun_like (F : Sort*) (α : out_param Sort*) (β : out_param $ α → Sort*)
+  [fun_like F α β] (p : F → Prop) : fun_like (subtype p) α β :=
+{ coe := fun_like.coe ∘ coe_subtype.coe,
+  coe_injective' := function.injective.comp fun_like.coe_injective' subtype.coe_injective }
+
```

Comment:
The following lemma works better than `coe_fn_coe_base'` in `combinatorics/derangements/basic.lean`, so let's try adding it. Hopefully it will solve some of the `coe_fn_coe_base` issues.
```suggestion

@[simp] lemma subtype.coe_fn_mk {F α : Type*} [has_coe_to_fun F (λ _, α)]
  {p : F → Prop} (f : F) (hf : p f) : (⇑(subtype.mk f hf) : F) = (⇑f : F) :=
rfl

```

Reply:
Are we sure we want this as a simp lemma? In general the subtype will have more properties than the base type, so getting rid of `hf` might not always be the best move.

Reply:
I think it's a good idea because it unblocks any `simp` lemmas for specific `f`, and any special properties of `hf` can still be added manually to the `simp` set. It also matches bundled morphism `coe_mk` lemmas such as `ring_hom.coe_mk`.

Reply:
You're right, I think I misread the statement.